tests/FileUploadFactoryTest.php

```
@@ -0,0 +1,258 @@
+    public function dummyMethodForMixedType(mixed $upload): void
+    {
+    }
+
+    public function dummyMethodForInvalidParameterType(string $invalidParam): void
+    {
+    }
+
+    /**
+     * Try to create a case with intersection types (PHP 8.1+)
+     */
+    public function dummyMethodForComplexUnionType((Traversable&Countable)|string|null $param): void
```

PHPMD (2.15.0)

Avoid unused parameters such as '$param'. (Unused Code Rules)

(UnusedFormalParameter)